Adding test coverage for server-mux.go by creating TestListenAndServeTLS()

jesselucas opened the pull request.

harshavardhana commented on cmd/config.go:
```diff
@@ -19,20 +19,27 @@ package cmd
 import (
 	"os"
```
Can your commit be separated into new lines? following an 80 column rule. 

It gets harder to read 'git logs'

harshavardhana replied:
Thanks for all the hardwork on this btw :+1:

jesselucas committed:
Adding test coverage for server-mux.go by creating TestListenAndServeTLS().
Editing config.go to add lock on customConfigPath to avoid race condition
when setGlobalConfigPath() and getConfigPath() are called concurrently.
